app/src/androidTest/java/com/github/se/di/module/TestModule.kt

```diff
@@ -158,7 +159,7 @@ object MockFirebaseFirestoreModule {
         .thenReturn(mockProfilesQuery)
     `when`(mockProfilesQuery.whereLessThanOrEqualTo(anyString(), anyString()))
         .thenReturn(mockProfilesQuery)
-    `when`(mockProfilesQuery.get()).thenReturn(mockTaskQuerySnapshotProfiles)
+    `when`(mockProfilesQuery.get(Source.SERVER)).thenReturn(mockTaskQuerySnapshotProfiles)
```

Comment:
Question : I'm curious, why do you have to add the Source.SERVER here ? What does it represent ?

Reply:
Basically that's why the offline mode was broken, it was removed by someone at some point. The idea is that for for our reconnection logic to launch, get() has to return onFailure, but firebase is well made and if there is no internet it gets the information from cache so it never launches the reconnection logic, by adding this you force firebase to get the information from the server which in turn, when fails, launches our reconnection logic with it's onFailure

Reply:
Ok it is clearer now, thank you !